addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -282,27 +282,35 @@ function factory(dependencies) {
             const moderated_channel_ids = this.thread.moderation
                 ? [this.thread.id]
                 : undefined;
-            const messages = await this.async(() =>
-                this.env.models['mail.message'].performRpcMessageFetch(
-                    domain,
-                    limit,
-                    moderated_channel_ids,
-                    context,
-                )
-            );
-            this.update({
-                fetchedMessages: [['link', messages]],
-                isLoaded: true,
-                isLoading: false,
-            });
-            if (!extraDomain && messages.length < limit) {
-                this.update({ isAllHistoryLoaded: true });
+            try {
+                const messages = await this.async(() =>
+                    this.env.models['mail.message'].performRpcMessageFetch(
+                        domain,
+                        limit,
+                        moderated_channel_ids,
+                        context,
+                    )
+                );
+                this.update({
+                    fetchedMessages: [['link', messages]],
+                    isLoaded: true,
+                    isLoading: false,
+                });
+                if (!extraDomain && messages.length < limit) {
+                    this.update({ isAllHistoryLoaded: true });
+                }
+                this.env.messagingBus.trigger('o-thread-cache-loaded-messages', {
+                    fetchedMessages: messages,
+                    threadCache: this,
+                });
+                return messages;
+            } catch(e) {
```

Comment:
How did it work before actually, when we had no catch? It was a silent error (in console uncaught in promise)?

Reply:
just nothing

Reply:
It's strange, if you are able to catch an exception it means there is one, how was it handled before then?

Reply:
The promise was rejected, so it was just silently failing. The try catch replace the .catch when using async/await